Description: allow mail command to grab the mail title from dep3 formalism
 If the patch is formatted according to
 http://dep.debian.net/deps/dep3/ then the mail subcommand will manage
 to extract a mail subject.

Index: quilt.git/quilt/mail.in
===================================================================
--- quilt.git.orig/quilt/mail.in	2012-12-19 15:23:42.659646504 +0100
+++ quilt.git/quilt/mail.in	2012-12-19 15:32:19.058893532 +0100
@@ -338,6 +338,19 @@
 			' $tmpdir/patch > $tmpdir/body
 		fi
 
+		# Does this patch have a Description: block?
+		if [ -z "$subject" ]
+		then
+			subject=$(formail -x Description: < $tmpdir/header |head -n 1)
+			if [ -n "$subject" ]
+			then
+				# Do not kill any headers as I don't really know what I want to kill:
+				# short version of Description was used as header, but not the long one,
+				# neither the other pseudo headers of the dep3 block
+				cp $tmpdir/patch $tmpdir/body
+			fi
+		fi
+
 		# Does this patch have DESC // subject // EDESC?
 		if [ -z "$subject" ]
 		then
@@ -442,10 +455,14 @@
 	subject=$(formail -x Replace-Subject: < "$tmpdir/$patch" | join_lines)
 	if [ $status -ne 0 -o -z "$subject" ]
 	then
-		printf \
+		subject=$(formail -x Description: < "$tmpdir/$patch" | head -n 1)
+		if [ $status -ne 0 -o -z "$subject" ]
+		then
+			printf \
 $"Unable to extract a subject header from %s\n" "$(print_patch "$patch")" >&2
-		rm -rf $tmpdir
-		exit 1
+			rm -rf $tmpdir
+			exit 1
+		fi
 	fi
 	subjects[${#subjects[@]}]="$patch"$'\t'"$subject"
 done
